### PR TITLE
Add subproject relationship between projects

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9323
https://github.com/cityofaustin/atd-data-tech/issues/9324
https://github.com/cityofaustin/atd-data-tech/issues/9310

## Testing
**URL to test:** 
https://moped-test.austinmobility.io/moped/projects/156

**Steps to test:**

1. Navigate to a project. As long as the project is not already a subproject/child project, you will see the Subprojects table. 
2. click the button to add a subproject
3. No parent projects nor already designated child projects should appear in the dropdown. Neither should the project you are viewing. 
4. Add a child project. 
5. Click child project link to confirm it takes you to that project's summary page. (There should not be a subproject table)
6. Delete a subproject. 

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryContractor.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ import { Box, Grid, Icon, TextField, Typography } from "@material-ui/core";
 import ProjectSummaryLabel from "./ProjectSummaryLabel";
 
 import {
-  PROJECT_UPDATE_CONTACTOR,
-  PROJECT_CLEAR_CONTACTOR,
+  PROJECT_UPDATE_CONTRACTOR,
+  PROJECT_CLEAR_CONTRACTOR,
```
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryContractor.js:8]: this was a typo

**Review by johnclary — APPROVED** [review]
amazing work @chiaberry! 

🛵 💨

**Review by amenity — APPROVED** [review]
(Speaking to the subprojects functionality itself; lists of projects on the "Projects" page and the My/Following lists of projects on the "Dashboard" page are 🔄 4ever, only completely loading intermittently )

...The feature itself works beautifully! Thank you, @chiaberry 🙏🏻 🚧 👪

**Review by mddilley — COMMENTED** [review]
This looks great! I was able to create subprojects locally and all the test steps worked out perfectly. 

I left one comment about removing `parent_project_id` from a child on delete migration. I'm not 100% that DB update works as expected.

File: moped-editor/src/views/projects/projectView/ProjectSummary/SubprojectsTable.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+} from "../../../../queries/subprojects";
+import typography from "../../../../theme/typography";
+
+const SubprojectsTable = ({ projectId = null }) => {
+  const addActionRef = React.useRef();
+
+  const { loading, error, data, refetch } = useQuery(SUBPROJECT_QUERY, {
+    variables: { projectId: projectId },
+    fetchPolicy: "no-cache",
+  });
```
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectSummary/SubprojectsTable.js:36]: I think that is right about the re-render. You've definitely got me thinking about how data is passed down to all the summary components. Looks like:

ProjectView fetches and then passes the whole data object to each child project tab
ProjectSummary tab gets data and then passes the whole object to each child summary component
There is also the refetch calls to think about which then trigger an update to data which cascades down through all these components again. 🤔
Another consideration for this specific query is if anything higher in the tree here needs to be aware of subprojects - probably not.

All that to say, I think this one-off request is okay but could fall under future refactoring.

File: moped-database/migrations/1655407355843_alter_table_public_moped_project_add_column_parent_project_id/up.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+alter table "public"."moped_project" add column "parent_project_id" integer
+ null;
+
+alter table "public"."moped_project"
+  add constraint "moped_project_parent_project_id_fkey"
+  foreign key ("parent_project_id")
+  references "public"."moped_project"
+  ("project_id") on update cascade on delete set null;
```
[the remark this answers is not in the document]
Comment (chiaberry) [reply moped-database/migrations/1655407355843_alter_table_public_moped_project_add_column_parent_project_id/up.sql:8]: Thank you @mddilley for pointing out the soft delete vs actual delete difference, I had forgotten the distinction. The soft delete will be handled here https://github.com/cityofaustin/atd-data-tech/issues/9329

[the remark this answers is not in the document]
Comment (mddilley) [reply moped-database/migrations/1655407355843_alter_table_public_moped_project_add_column_parent_project_id/up.sql:8]: Oh awesome, that makes sense - approving! ✅

**Review by mddilley — APPROVED** [review]

**Comment by chiaberry** [comment]
I am going to merge this to staging so Rebecca can take a look at it since we are still having problems with her account on test. (and test is generally being annoying) 

I will address some of the feedback and refactoring in my next branch for subprojects, https://github.com/cityofaustin/atd-data-tech/issues/9330

File: moped-database/migrations/1655407355843_alter_table_public_moped_project_add_column_parent_project_id/up.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (johnclary) [reply moped-database/migrations/1655407355843_alter_table_public_moped_project_add_column_parent_project_id/up.sql:8]: a bit of related background for you [here](https://github.com/cityofaustin/atd-moped/pull/539) @mddilley - until Frank made those changes to the cascading FKs, it was really difficult to hard delete Moped projects, should the need arise—as it did when we were migrating AMD's Data Tracker projects into Moped.

[the remark this answers is not in the document]
Comment (johnclary) [reply moped-database/migrations/1655407355843_alter_table_public_moped_project_add_column_parent_project_id/up.sql:8]: so the general philosophy is to allow deletes at the DB level, and assume that a dev would be doing it for good reason.

**Pull request merged**